Added documentation for prerequisites to Running Spec Tests Locally (#517)
opensearch-project/opensearch-api-specification: TESTING_GUIDE.md

- [Spec Testing Guide](#spec-testing-guide)
  - [Running Spec Tests](#running-spec-tests)
    - [Running Spec Tests Locally](#running-spec-tests-locally)
      - [Prerequisites](#prerequisites)
      - [OpenSearch Cluster](#opensearch-cluster)

>       - [Run Tests ](#run-tests)

Reviewer: There's an extra space after Run Tests.

```suggestion
      - [Run Tests](#run-tests)
```

### Running Spec Tests Locally

> Downloading and installing latest version of Node.js and npm

Reviewer: ```suggestion
Download and install the latest version of Node.js and npm from [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
```

Author: ok

### Running Spec Tests Locally

> Prerequisites: 

Reviewer: Make this an actual section.

```suggestion
#### Prerequisites 
```

Add a section below:

```
#### Running Tests
```

> Download and install latest version of Node.js and npm from 

Reviewer: ```suggestion
Download and install the latest versions of Node.js and npm from [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
```

[https://docs.npmjs.com/downloading-and-installing-node-js-and-npm]
```bash

> Note: to download the latest version of npm, on the command line, run the following command:

Reviewer: Remove this note. If you don't have `npm` you cannot run `npm -g`. Running it again with `npm -g npm` is not useful either. It's not our business here to explain how to upgrade npm.

npm install -g npm
```

> In Windows using PowerShell:

Reviewer: Since tests don't work on Windows, remove the part about Windows for now. Once you've fixed tests to run on Windows we can add this.

### Running Spec Tests Locally

Download and install latest version of Node.js and npm from [here]

> https://docs.npmjs.com/downloading-and-installing-node-js-and-npm

Reviewer: To make "here" a link the syntax is "[here](link)", so:

```suggestion
Download and install the latest version of Node.js and npm from [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
```

Reviewer: Hit preview, this renders the following:

<img width="1068" alt="Screenshot 2024-08-21 at 10 02 18 AM" src="https://github.com/user-attachments/assets/f4206b6f-7600-4657-ab58-f2e6dc6cb8d6">

"here" needs to be a working link.

### Running Spec Tests Locally

#### Prerequisites 

> Download and install the latest version of Node.js and npm from [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) and run 'npm install'.

Reviewer: Use backtick quotes so that `npm install` renders.

```suggestion
Download and install the latest version of Node.js and npm from [here](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) and run `npm install`.
```

cd tests/default
docker compose up -d
```

> In Windows env:

Reviewer: ```suggestion
In Windows using PowerShell:
```

Author: ok

``` 

> $Env:OPENSEARCH_PASSWORD="MyPassword!"

Reviewer: ```suggestion
$Env:OPENSEARCH_PASSWORD="<<your password>>"
```

To match the one above. Or replace <<your password>> with the default password we have `MyStrongPassword123!`.

Author: ok

cd tests/default
docker compose up -d
```

> #### Run Tests 

Reviewer: There's an extra space after `Run Tests `.

```suggestion
#### Run Tests
```
